libcudacxx/test/libcudacxx/std/containers/sequences/inplace_vector/assign.pass.cpp

```diff
@@ -0,0 +1,353 @@
+    input_iter_iter_empty.assign(iter{expected.begin()}, iter{expected.end()});
+    assert(input_iter_iter_empty.empty());
+  }
+
+  {
+    using iter                            = cpp17_input_iterator<const T*>;
+    const cuda::std::array<T, 0> expected = {};
+    cuda::std::inplace_vector<T, 42> input_iter_iter_shrink_empty{T(1), T(42), T(1337), T(0)};
+    input_iter_iter_shrink_empty.assign(iter{expected.begin()}, iter{expected.end()});
+    assert(input_iter_iter_shrink_empty.empty());
+  }
+
+  {
+    using iter                            = cpp17_input_iterator<const T*>;
+    const cuda::std::array<T, 2> expected = {T(42), T(42)};
+    cuda::std::inplace_vector<T, 42> input_iter_iter_shrink{T(1), T(42), T(1337), T(0)};
+    input_iter_iter_shrink.assign(iter{expected.begin()}, iter{expected.end()});
+    assert(!input_iter_iter_shrink.empty());
+    assert(
+      cuda::std::equal(input_iter_iter_shrink.begin(), input_iter_iter_shrink.end(), expected.begin(), expected.end()));
+  }
+
+  {
+    using iter                            = cpp17_input_iterator<const T*>;
+    const cuda::std::array<T, 6> expected = {T(42), T(42), T(42), T(42), T(42), T(42)};
+    cuda::std::inplace_vector<T, 42> input_iter_iter_grow{T(1), T(42), T(1337), T(0)};
+    input_iter_iter_grow.assign(iter{expected.begin()}, iter{expected.end()});
+    assert(!input_iter_iter_grow.empty());
+    assert(
+      cuda::std::equal(input_iter_iter_grow.begin(), input_iter_iter_grow.end(), expected.begin(), expected.end()));
+  }
+
+  {
+    const cuda::std::array<T, 0> expected = {};
+    cuda::std::inplace_vector<T, 42> iter_iter_empty{};
+    iter_iter_empty.assign(expected.begin(), expected.end());
+    assert(iter_iter_empty.empty());
+  }
+
+  {
+    const cuda::std::array<T, 0> expected = {};
+    cuda::std::inplace_vector<T, 42> iter_iter_shrink_empty{T(1), T(42), T(1337), T(0)};
+    iter_iter_shrink_empty.assign(expected.begin(), expected.end());
+    assert(iter_iter_shrink_empty.empty());
+  }
+
+  {
+    const cuda::std::array<T, 2> expected = {T(42), T(42)};
+    cuda::std::inplace_vector<T, 42> iter_iter_shrink{T(1), T(42), T(1337), T(0)};
+    iter_iter_shrink.assign(expected.begin(), expected.end());
+    assert(!iter_iter_shrink.empty());
+    assert(cuda::std::equal(iter_iter_shrink.begin(), iter_iter_shrink.end(), expected.begin(), expected.end()));
+  }
+
+  {
+    const cuda::std::array<T, 6> expected = {T(42), T(42), T(42), T(42), T(42), T(42)};
+    cuda::std::inplace_vector<T, 42> iter_iter_grow{T(1), T(42), T(1337), T(0)};
+    iter_iter_grow.assign(expected.begin(), expected.end());
+    assert(!iter_iter_grow.empty());
+    assert(cuda::std::equal(iter_iter_grow.begin(), iter_iter_grow.end(), expected.begin(), expected.end()));
+  }
+
+  {
+    const cuda::std::initializer_list<T> expected = {};
+    cuda::std::inplace_vector<T, 42> init_list_empty{};
+    init_list_empty.assign(expected);
+    assert(init_list_empty.empty());
```

Comment:
Yeah, I think your vector names are meant to describe the purpose of the test. `init_list_empty` is not a great name for the vector under test IMO.

Suggestion: Please make that a comment.

Reply:
Adopted to a simpler naming sheme,. thanks a lot